benchmark/bench_test.go

```diff
@@ -0,0 +1,257 @@
+	},
+}
+
+var CacheCases = map[string]uint32{
+	"no_cache": 0,
+	"cache":    1,
+}
+
+var PrepareGasLimit uint64 = 4000000
+var ExecuteGasLimit uint64 = 4000000
+
+// benchmark test for prepare function of owasm vm
+func BenchmarkOwasmVMPrepare(b *testing.B) {
+	for cache, cacheSize := range CacheCases {
+		for name, tc := range PrepareCases {
+			for _, pm := range tc.parameters {
+				b.Run(fmt.Sprintf(
+					"%s - %s (param: %d)",
+					cache,
+					name,
+					pm,
+				), func(b *testing.B) {
+					// prepare owasm vm
+					owasmVM, err := owasm.NewVm(cacheSize)
+					require.NoError(b, err)
+
+					// prepare owasm code
+					oCode, err := GetBenchmarkWasm()
+					require.NoError(b, err)
+					compiledCode, err := owasmVM.Compile(oCode, types.MaxCompiledWasmCodeSize)
+					require.NoError(b, err)
+
+					// prepare request
+					req := types.NewRequest(
+						1, obi.MustEncode(BenchmarkCalldata{
+							DataSourceId: 1,
+							Scenario:     tc.scenario,
+							Value:        pm,
+						}), []sdk.ValAddress{}, 1,
+						1, time.Now(), "", nil, nil, ExecuteGasLimit,
+					)
+
+					b.ResetTimer()
+					b.StopTimer()
+
+					// call prepare on new env
+					for i := 0; i < b.N; i++ {
+						env := types.NewPrepareEnv(
+							req,
+							int64(types.DefaultMaxCalldataSize),
+							int64(types.DefaultMaxRawRequestCount),
+						)
+						b.StartTimer()
+						_, _ = owasmVM.Prepare(
+							compiledCode,
+							oraclekeeper.ConvertToOwasmGas(PrepareGasLimit),
+							int64(types.DefaultMaxCalldataSize),
+							env,
+						)
+						b.StopTimer()
+					}
+				})
+			}
+		}
+	}
+}
+
+// benchmark test for execute function of owasm vm
+func BenchmarkOwasmVMExecute(b *testing.B) {
+	for cache, cacheSize := range CacheCases {
+		for name, tc := range ExecuteCases {
+			for _, pm := range tc.parameters {
+				b.Run(fmt.Sprintf(
+					"%s - %s (param: %d)",
+					cache,
+					name,
+					pm,
+				), func(b *testing.B) {
+					// prepare owasm vm
+					owasmVM, err := owasm.NewVm(cacheSize)
+					require.NoError(b, err)
+
+					// prepare owasm code
+					oCode, err := GetBenchmarkWasm()
+					require.NoError(b, err)
+					compiledCode, err := owasmVM.Compile(oCode, types.MaxCompiledWasmCodeSize)
+					require.NoError(b, err)
+
+					// prepare request
+					req := types.NewRequest(
+						1, obi.MustEncode(BenchmarkCalldata{
+							DataSourceId: 1,
+							Scenario:     tc.scenario,
+							Value:        pm,
+						}), []sdk.ValAddress{}, 1,
+						1, time.Now(), "", nil, nil, ExecuteGasLimit,
+					)
+
+					b.ResetTimer()
+					b.StopTimer()
+
+					// call execute on new env
+					for i := 0; i < b.N; i++ {
+						env := types.NewExecuteEnv(req, []types.Report{}, time.Now())
+
+						b.StartTimer()
+						_, _ = owasmVM.Execute(
+							compiledCode,
+							oraclekeeper.ConvertToOwasmGas(ExecuteGasLimit),
+							int64(types.DefaultMaxCalldataSize),
+							env,
+						)
+						b.StopTimer()
+					}
+				})
+			}
+		}
+	}
+}
```

Comment:
ditto

Reply:
done, create setup function for owasm env.